app/src/main/java/dev/andrylat/app/App.java

```diff
@@ -0,0 +1,155 @@
+package dev.andrylat.app;
+
+import java.util.Scanner;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.IntStream;
+
+public class App {
+    private StringBuilder errors;
+    private Scanner scanner;
+    private String brand = "";
+    private static final int validCreditCardNumberLength = 16;
+
+    public App() {
+        System.out.println("Hello. Enter card number for validation:");
+        errors = new StringBuilder();
+        scanner = new Scanner(System.in);
+    }
+
+    public static void main(String[] args) {
+        new App().process().print();
+    }
+
+    public App process() {
+        var input = scanner.nextLine();
+        
+        validate(input);
+
+        var cardNumber = getCardNumber(input);
+        determineBrandByNumber(cardNumber);
+        validate(cardNumber);
+
+        return this;
+    }
+
+    private void determineBrandByNumber(List<Integer> cardNumber) {
+        /*
+           for later improvements:
+
+           I could create an enum with all card brands,
+           and each card brand would have a method to validate
+           if the given number equals to the brand's possible number
+           the return type of the method would be boolean
+
+        */
+
+        switch (cardNumber.get(0)) {
+            case 4: brand = "VISA";
+            break;
+            case 5: brand = IntStream.range(0,6).anyMatch(n -> n == cardNumber.get(1)) ? "MASTERCARD" : "";
+            break;
+        }
+    }
+
+    private List<Integer> getCardNumber(String s) {
+        return new ArrayList<>(Arrays.stream(s.split(""))
+                         .filter(c -> !c.equals(" "))
+                         .filter(this::isDigit)
+                         .map(Integer::valueOf)
+                         .toList());
+    }
+
+    private void validate(String input) {
+        if (input.length() < validCreditCardNumberLength) {
+            errors.append("-> Length should be ")
+                  .append(validCreditCardNumberLength)
+                  .append(" symbols\n");
+        }
+    }
+
+    private void validate(List<Integer> cardNumber) {
```

Comment:
you usse the same name but with different params. When you read the code, is it clear for you what is validated where? You must use proper names, which will describe what is going on!